src/Symfony/Component/Serializer/Tests/SerializerTest.php

```diff
@@ -1725,6 +1726,19 @@ public function testDenormalizationFailsWithMultipleErrorsInDefaultContext()
             $this->assertSame($expected, $exceptionsAsArray);
         }
     }
+
+    /**
+     * @requires PHP 8.4
+     */
+    public function testDeserializeObjectWithAsymmetricPropertyVisibility()
+    {
+        $serializer = new Serializer([new ObjectNormalizer()], ['json' => new JsonEncoder()]);
+        /** @var AsymmetricVisibilityDummy $object */
+        $object = $serializer->deserialize(json_encode(['type' => 'final', 'item' => 'one']), AsymmetricVisibilityDummy::class, 'json');
+
+        $this->assertSame('one', $object->item);
+        $this->assertSame('final', $object->type);
```

Comment:
the values in the constructor and in the test case are the same, which means we don't know what we test